File: backend-server/internal/model/storage_config.go
package model

import "time"

// StorageConfig 存储连接配置模型
// 用于管理多个同类型的存储连接配置（如多个 MinIO、多个 OSS 等）
type StorageConfig struct {
	ID          int64     `json:"id" gorm:"primaryKey;autoIncrement"`
	Name        string    `json:"name" gorm:"size:100;not null;uniqueIndex:uk_name"`
	Driver      string    `json:"driver" gorm:"size:20;not null;index:idx_driver"`
	Endpoint    string    `json:"endpoint" gorm:"size:500"`
	AccessKey   string    `json:"accessKey" gorm:"size:500"`
	SecretKey   string    `json:"secretKey" gorm:"size:500"`
	Bucket      string    `json:"bucket" gorm:"size:200"`
	Region      string    `json:"region" gorm:"size:100"`
	UseSSL      bool      `json:"useSsl" gorm:"default:false"`
	CDNDomain   string    `json:"cdnDomain" gorm:"size:500"`
	IsDefault           bool      `json:"isDefault" gorm:"default:false"`
	PresignedURLExpiry  int       `json:"presignedUrlExpiry" gorm:"not null;default:3600"`
	Status              int8      `json:"status" gorm:"default:1;index:idx_status"`
	Description string    `json:"description" gorm:"size:500"`
	CreatedAt   time.Time `json:"createdAt" gorm:"autoCreateTime"`
	UpdatedAt   time.Time `json:"updatedAt" gorm:"autoUpdateTime"`
}

func (StorageConfig) TableName() string {
	return "sys_storage_config"
}

File: frontend-admin/apps/web-antd/src/api/system/storage-config.ts
import { requestClient } from '#/api/request';

export namespace StorageConfigApi {
  export interface StorageConfig {
    id: number;
    name: string;
    driver: string;
    endpoint: string;
    accessKey: string;
    secretKey: string;
    bucket: string;
    region: string;
    useSsl: boolean;
    cdnDomain: string;
    isDefault: boolean;
    presignedUrlExpiry: number;
    status: number;
    description: string;
    createdAt: string;
    updatedAt: string;
  }

  export interface CreateStorageConfig {
    name: string;
    driver: string;
    endpoint?: string;
    accessKey?: string;
    secretKey?: string;
    bucket?: string;
    region?: string;
    useSsl?: boolean;
    cdnDomain?: string;
    isDefault?: boolean;
    presignedUrlExpiry?: number;
    status?: number;
    description?: string;
  }

  export interface TestConnectionData {
    driver: string;
    endpoint?: string;
    accessKey?: string;
    secretKey?: string;
    bucket?: string;
    region?: string;
    useSsl?: boolean;
  }
}

/** 获取所有存储配置 */
export async function getAllStorageConfigsApi() {
  return requestClient.get<StorageConfigApi.StorageConfig[]>(
    '/system/storage-configs',
  );
}

/** 根据ID获取存储配置 */
export async function getStorageConfigByIdApi(id: number) {
  return requestClient.get<StorageConfigApi.StorageConfig>(
    `/system/storage-configs/${id}`,
  );
}

/** 创建存储配置 */
export async function createStorageConfigApi(
  data: StorageConfigApi.CreateStorageConfig,
) {
  return requestClient.post<StorageConfigApi.StorageConfig>(
    '/system/storage-configs',
    data,
  );
}

/** 更新存储配置 */
export async function updateStorageConfigApi(
  id: number,
  data: StorageConfigApi.CreateStorageConfig,
) {
  return requestClient.put<StorageConfigApi.StorageConfig>(
    `/system/storage-configs/${id}`,
    data,
  );
}

/** 删除存储配置 */
export async function deleteStorageConfigApi(id: number) {
  return requestClient.delete(`/system/storage-configs/${id}`);
}

/** 设置默认存储配置 */
export async function setDefaultStorageConfigApi(id: number) {
  return requestClient.put(`/system/storage-configs/${id}/default`);
}

/** 测试已有配置的连接 */
export async function testStorageConfigConnectionApi(id: number) {
  return requestClient.post<{ connected: boolean }>(
    `/system/storage-configs/${id}/test`,
  );
}

/** 根据数据测试连接 */
export async function testStorageConfigByDataApi(
  data: StorageConfigApi.TestConnectionData,
) {
  return requestClient.post<{ connected: boolean }>(
    '/system/storage-configs/test-by-data',
    data,
  );
}

/** 获取已启用的驱动列表 */
export async function getStorageConfigEnabledDriversApi() {
  return requestClient.get<
    Array<{ value: string; label: string; icon: string; enabled: boolean }>
  >('/system/storage-configs/enabled-drivers');
}
